src/types/privatekubectl.ts

```diff
@@ -93,23 +106,32 @@ export class PrivateKubectl extends Kubectl {
    }
 
    public extractFilesnames(strToParse: string) {
-      let start = strToParse.indexOf('-filename')
-      let offset = 7
-
-      if (start == -1) {
-         start = strToParse.indexOf('-f')
-
-         if (start == -1) {
-            return ''
+      const fileNames: string[] = []
+      const argv = require('minimist')(strToParse.split(' '))
+      const fArg = 'f'
+      const filenameArg = 'filename'
+
+      if (argv[fArg]) {
```

Comment:
can we abstract this logic into a method that runs on both the args instead of having two copies of the same logic right after each other?